pulp_2_tests/tests/rpm/api_v2/test_duplicate_uploads.py

```diff
@@ -86,3 +88,97 @@ def do_test(self, feed, type_id, body, unit_key=None):
+           the upload and import was successful.
+        3. Copy the content to a target repo. Assert the copy was successful.
+        4. Upload identically named but different content and import it into
+           the source repository. Assert the upload and import was successful.
+        5. Copy the content to the target repo.
+        6. Assert the target manifest only has one entry for the content.
+        """
+        cfg = config.get_config()
+        if cfg.pulp_version < Version('2.20'):
+            raise unittest.SkipTest('This test requires Pulp 2.20 or newer.')
+        client = api.Client(cfg, api.json_handler)
+
+        # 1. Create two iso-repo
+        repos = []
+        iso = utils.http_get(FILE_URL)
+        unit_key = {
+            'checksum': hashlib.sha256(iso).hexdigest(),
+            'name': os.path.basename(urlsplit(FILE_URL).path),
+            'size': len(iso),
+        }
+        data = {
+            'importer_type_id': 'iso_importer',
+            'notes': {'_repo-type': 'iso-repo'},
+        }
+        repos.append(client.post(REPOSITORY_PATH, gen_repo(**data)))
+        self.addCleanup(client.delete, repos[0]['_href'])
+        repos.append(client.post(REPOSITORY_PATH, gen_repo(**data)))
+        self.addCleanup(client.delete, repos[1]['_href'])
+
+        # 2. Import the units into the source repo and verify
+        call_report = upload_import_unit(cfg, iso, {
+            'unit_type_id': 'iso',
+            'unit_key': unit_key
+        }, repos[0])
+        self.assertIsNone(call_report['result'])
+
+        # Only one iso unit should exist in the source repo
+        self.assertEqual(
+            len(search_units(cfg, repos[0], {'type_ids': ['iso']})),
+            1,
+            search_units(cfg, repos[0], {'type_ids': ['iso']})
+        )
+
+        # 3. Sync to a target repository.
+        client.post(urljoin(repos[1]['_href'], 'actions/associate/'), {
+            'source_repo_id': repos[0]['id'],
+            'override_config': {},
+            'criteria': {'filters': {'unit': {}}, 'type_ids': ['iso']},
+        })
+
+        # Assert that the single ISO was copied.
+        self.assertEqual(
+            len(search_units(cfg, repos[1], {'type_ids': ['iso']})),
+            1,
+            search_units(cfg, repos[1], {'type_ids': ['iso']})
+        )
+
+        # 4. Upload a same-name, but different ISO to the source repo
+        iso2 = utils.http_get(FILE2_URL)
+        unit_key = {
+            'checksum': hashlib.sha256(iso2).hexdigest(),
+            'name': os.path.basename(urlsplit(FILE_URL).path),
+            'size': len(iso2),
+        }
+        call_report = upload_import_unit(cfg, iso2, {
+            'unit_type_id': 'iso',
+            'unit_key': unit_key
+        }, repos[0])
+        self.assertIsNone(call_report['result'])
+
+        # 5. Copy the new iso from the source to target repo
+        client.post(urljoin(repos[1]['_href'], 'actions/associate/'), {
```

Comment:
We have a `sync` function that takes a remote, in this case I see this `/associate/` used twice, maybe we can create a helper function for it? `pulp2.utils.associate` ?

Reply:
@rochacbruno that may be useful. 

Before creating my two local repos to sync against, I looked to see how similar repo-to-repo copies were being done. It appears, for differing `type_ids` we are doing the same thing a lot with varying criteria. The search showed `19 matches across 11 files` in the repo from docker to rpm.

I will look into this afternoon and make something useful for my case and make it somewhat extensible for the cases already there.

Do we want to consider a refactor issue for those doing the same or is that not worth the time investment?

Reply:
@bherrin3 we can merge this one as it is and then open a new issue to track the refactor of all the other uses of it.